crates/runtime-table-partition/src/creator/filename.rs

```diff
@@ -27,40 +27,65 @@ limitations under the License.
 //! on variants that have types that can be serialized/deserialized.
 
 use arrow_schema::DataType;
+use datafusion::prelude::Expr;
 use datafusion::scalar::ScalarValue;
 use serde::{Deserialize, Serialize};
 use snafu::prelude::*;
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
 
 #[derive(Debug, Snafu)]
 pub enum Error {
-    #[snafu(display("Failed to serialize scalar value: {source}"))]
+    #[snafu(display("Failed to serialize: {source}"))]
     Serialize { source: serde_qs::Error },
 
-    #[snafu(display("Failed to deserialize scalar value: {source}"))]
+    #[snafu(display("Failed to deserialize: {source}"))]
     Deserialize { source: serde_qs::Error },
 
     #[snafu(display("Unsupported scalar value type: {data_type}"))]
     UnsupportedType { data_type: DataType },
 }
 
-/// Converts a [`ScalarValue`] to its [`String`] representation.
+#[derive(Serialize, Deserialize)]
+struct SerializablePair {
+    scalar: SupportedScalarValue,
+    exprs_hash: String, // Store hash of Vec<Expr> Debug string
+}
+
+/// Hashes the Debug string of a [`Vec<Expr>`] using `DefaultHasher`.
+#[must_use]
+pub fn hash_exprs(exprs: &[Expr]) -> String {
+    let exprs_str = format!("{exprs:?}");
```

Comment:
I'm not sure that this is guaranteed to be stable across version upgrades. It would be unfortunate if a DataFusion upgrade broke existing partitioned tables.

Reply:
I've instead hashed the original Strings from the config file. Let me know if you think that is problematic. I was trying to avoid building out the metadata handling in the checkpointing database now.

Reply:
Looks good, thanks!